server/settings/settings.go

```diff
@@ -61,7 +61,14 @@ func (s *Server) Get(ctx context.Context, q *settingspkg.SettingsQuery) (*settin
 			break
 		}
 	}
-
+	configManagementPlugins, err := s.mgr.GetConfigManagementPlugins()
+	if err != nil {
+		return nil, err
+	}
+	tools := make([]*v1alpha1.ConfigManagementPlugin, len(configManagementPlugins))
```

Comment:
I think we should not expose config management plugin init/generate commands since /settings API is available without authentication.

We can simplify check if request has valid token and remove init/generate command details from the tool if token is missing.

You can use the same technique as session.go API:

modify AuthFuncOverride:

https://github.com/argoproj/argo-cd/blob/master/server/session/session.go#L57

use session manager to check if user is logged in:

https://github.com/argoproj/argo-cd/blob/master/server/session/session.go#L65

Reply:
Thank you for the feedback! Change is checked in.